### PR TITLE
[Backup] BugFix: Enabled Alternate Location Restore for SQL/SAPHANA protected items

### PR DESCRIPTION
**Pull request by sambitratha**
**Description**
Enabled Alternate Location Restore for SQL/SAPHANA protected items

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Backup

**Comment by yungezz** [comment]
hi @qwordy could  you pls help to review? thanks

File: src/azure-cli/azure/cli/command_modules/backup/custom_base.py [hunk 1]
```diff
@@ -402,9 +402,14 @@ def show_recovery_config(cmd, client, resource_group_name, vault_name, restore_m
     target_item = None
     if target_item_name is not None:
         protectable_items_client = backup_protectable_items_cf(cmd.cli_ctx)
-        target_item = show_protectable_instance(cmd, protectable_items_client, resource_group_name, vault_name,
-                                                target_server_name, target_server_type,
-                                                workload_type, container_name)
+        if target_container_name is None:
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/backup/custom_base.py:405]: the change is ok, while code could be refined.

**Review by yungezz — APPROVED** [review]

**Comment by qwordy** [comment]
Code owner review required

**Comment by sambitratha** [comment]
Hi @fengzhou-msft  could you please review this PR?

File: src/azure-cli/azure/cli/command_modules/backup/custom_base.py [hunk 2]
```diff
@@ -402,9 +402,14 @@ def show_recovery_config(cmd, client, resource_group_name, vault_name, restore_m
...
+            target_item = show_protectable_instance(cmd, protectable_items_client, resource_group_name, vault_name,
+                                                    target_server_name, target_server_type,
+                                                    workload_type, container_name)
+        else:
+            target_item = show_protectable_instance(cmd, protectable_items_client, resource_group_name, vault_name,
+                                                    target_server_name, target_server_type,
+                                                    workload_type, target_container_name)
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/backup/custom_base.py:412]:
```suggestion
        target_item = show_protectable_instance(cmd, protectable_items_client, resource_group_name, vault_name,
                                                target_server_name, target_server_type,
                                                workload_type,
                                                container_name if target_container_name is None else \
                                                target_container_name)
```

**Comment by sambitratha** [comment]
Hi @fengzhou-msft  i've resolved your comment. please restart the azure pipelines and approve. Thanks

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**